app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -37,17 +36,26 @@ def save_ems_physical_infra_inventory(ems, hashes, target = nil)
     ems
   end
 
+  def save_physical_racks_inventory(ems, hashes, target = nil)
+    target = ems if target.nil?
+
+    deletes = target == ems ? :use_association : []
+    ems.physical_racks.reset
+    save_inventory_multi(ems.physical_racks, hashes, deletes, [:ems_ref])
+    store_ids_for_new_records(ems.physical_racks, hashes, :ems_ref)
+  end
+
   def save_physical_servers_inventory(ems, hashes, target = nil)
     target = ems if target.nil?
 
     ems.physical_servers.reset
-    deletes = if target == ems
-                :use_association
-              else
-                []
-              end
+    deletes = target == ems ? :use_association : []
+
+    child_keys = %i(computer_system asset_detail hosts)
+    hashes.each do |h|
+      h[:physical_rack_id] = h.delete(:rack).try(:[], :id)
```

Comment:
Now we're talking :D

Reply:
Sorry one last nit-pick then this looks good to me, the hash key here `:rack` should match the fk name so it should be `:physical_rack`

Reply:
Done (: